mlir/lib/Dialect/PDL/IR/Builtins.cpp

```diff
@@ -51,52 +52,172 @@ mlir::Attribute addElemToArrayAttr(mlir::PatternRewriter &rewriter,
   return rewriter.getArrayAttr(values);
 }
 
-LogicalResult add(mlir::PatternRewriter &rewriter, mlir::PDLResultList &results,
-                  llvm::ArrayRef<mlir::PDLValue> args) {
-  assert(args.size() == 2 && "Expected 2 arguments");
+template <UnaryOpKind T>
+LogicalResult unaryOp(PatternRewriter &rewriter, PDLResultList &results,
+                      ArrayRef<PDLValue> args) {
+  assert(args.size() == 1 && "Expected one operand for unary operation");
+  auto operandAttr = args[0].cast<Attribute>();
+
+  if (auto operandIntAttr = dyn_cast_or_null<IntegerAttr>(operandAttr)) {
+    auto integerType = cast<IntegerType>(operandIntAttr.getType());
+    auto bitWidth = integerType.getIntOrFloatBitWidth();
+
+    if constexpr (T == UnaryOpKind::exp2) {
+      uint64_t resultVal =
+          integerType.isUnsigned() || integerType.isSignless()
+              ? std::pow(2, operandIntAttr.getValue().getZExtValue())
```

Comment:
Is this operator what we need actually? Not for this PR but wouldn't it better to have pow instead of this? or even `exp`?